zipkin-lens/src/components/DependenciesPage/DependenciesGraph.tsx

```diff
@@ -0,0 +1,312 @@
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return object.source.name === value || object.target.name === value;
+};
+
+const filterNodes = (object: any, value: any) => {
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return (
+    object.incomingConnections.find(
+      (conn: any) => conn.source.name === value,
+    ) ||
+    object.outgoingConnections.find((conn: any) => conn.target.name === value)
+  );
+};
+
+// Export for testing.
+export const getNodesAndEdges = (dependencies: Dependencies) => {
+  const nodes = [] as { name: string }[];
+  const edges = [] as Edge[];
+
+  dependencies.forEach((edge) => {
+    const nodeNames = nodes.map((node) => node.name);
+
+    if (!nodeNames.includes(edge.parent)) {
+      nodes.push({ name: edge.parent });
+    }
+    if (!nodeNames.includes(edge.child)) {
+      nodes.push({ name: edge.child });
+    }
+
+    edges.push({
+      source: edge.parent,
+      target: edge.child,
+      metrics: {
+        normal: edge.callCount || 0,
+        danger: edge.errorCount || 0,
+      },
+    });
+  });
+  return { nodes, edges };
+};
+
+const reactSelectStyles = {
+  control: (base: CSSProperties) => ({
+    ...base,
+    width: '15rem',
+  }),
+  option: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+  clearIndicator: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+  dropdownIndicator: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    containerGrid: {
+      height: '100%',
+    },
+    graphItemGrid: {
+      height: '100%',
+    },
+    nodeDetailDataItemGrid: {
+      height: '100%',
+      boxShadow: theme.shadows[10],
+      zIndex: 1,
+    },
+    vizceralWrapper: {
+      // In Firefox, 100% height does not work...
+      '& .vizceral': {
+        height: '99%',
+      },
+    },
+  }),
+);
+
+interface Props {
+  dependencies: Dependencies;
+}
+
+const DependenciesGraph: React.FC<Props> = ({ dependencies }) => {
+  const classes = useStyles();
+  const theme = useTheme();
+  const vizStyle = {
+    colorText: theme.palette.primary.contrastText,
+    colorTextDisabled: theme.palette.primary.contrastText,
+    colorConnectionLine: theme.palette.grey[800],
+    colorTraffic: {
+      normal: theme.palette.primary.dark,
+      warning: theme.palette.secondary.dark,
+      danger: theme.palette.secondary.dark,
+    },
+    colorDonutInternalColor: theme.palette.primary.light,
+    colorDonutInternalColorHighlighted: theme.palette.primary.light,
+    colorLabelBorder: theme.palette.primary.dark,
+    colorLabelText: theme.palette.primary.contrastText,
+    colorTrafficHighlighted: {
+      normal: theme.palette.primary.main,
+    },
+  };
+
+  const [focusedNodeName, setFocusedNodeName] = React.useState('');
```

Comment:
Import the methods from `React`?

Reply:
These days I prefer using `React.useState` over `import { useState } from 'react'`.
The reason for this is that if `useState` is no longer used in this file in the future, I won't have to check and change the `import` statement.

Reply:
I sort of understand the motivation, but it doesn't seem strong enough to use a pattern different from every tutorial on hooks on the web, including official one. We want to make sure the code is accessible to any contributor.

Reply:
OK!